libraries/AP_NavEKF/EKFGSF_yaw.cpp

```diff
@@ -0,0 +1,678 @@
+	if (vel_data_updated && run_ekf_gsf) {
+		if (!vel_fuse_running) {
+			// Perform in-flight alignment
+			for (uint8_t mdl_idx = 0; mdl_idx < N_MODELS_EKFGSF; mdl_idx ++) {
+				// Use the firstGPS  measurement to set the velocities and corresponding variances
+				EKF[mdl_idx].X[0] = vel_NE[0];
+				EKF[mdl_idx].X[1] = vel_NE[1];
+				EKF[mdl_idx].P[0][0] = sq(fmaxf(vel_accuracy, 0.5f));
+				EKF[mdl_idx].P[1][1] = EKF[mdl_idx].P[0][0];
+			}
+			alignYaw();
+			vel_fuse_running = true;
+		} else {
+			float total_w = 0.0f;
+			float newWeight[(uint8_t)N_MODELS_EKFGSF];
+			for (uint8_t mdl_idx = 0; mdl_idx < N_MODELS_EKFGSF; mdl_idx ++) {
+				// Update states and covariances using GPS NE velocity measurements fused as direct state observations
+				correct(mdl_idx);
+
+				// Calculate weighting for each model assuming a normal distribution
+				newWeight[mdl_idx]= fmaxf(gaussianDensity(mdl_idx) * GSF.weights[mdl_idx], 0.0f);
+				total_w += newWeight[mdl_idx];
+			}
+
+			// Normalise the sume of weights to unity
+			if (vel_fuse_running && total_w > 0.0f) {
+				float total_w_inv = 1.0f / total_w;
+				for (uint8_t mdl_idx = 0; mdl_idx < N_MODELS_EKFGSF; mdl_idx ++) {
+					GSF.weights[mdl_idx]  = newWeight[mdl_idx] * total_w_inv;
+				}
+			}
+		}
+	} else if (vel_fuse_running && !run_ekf_gsf) {
+		// We have landed so reset EKF-GSF states and wait to fly again
+		// Do not reset the AHRS as it continues to run when on ground
+		initialise();
+		vel_fuse_running = false;
+	}
+
+	// Calculate a composite state vector as a weighted average of the states for each model.
+	// To avoid issues with angle wrapping, the yaw state is converted to a vector with legnth
+	// equal to the weighting value before it is summed.
+	memset(&GSF.state, 0, sizeof(GSF.state));
+	Vector2f yaw_vector = {};
+	for (uint8_t mdl_idx = 0; mdl_idx < N_MODELS_EKFGSF; mdl_idx ++) {
+		for (uint8_t state_index = 0; state_index < 2; state_index++) {
+			GSF.state[state_index] += EKF[mdl_idx].X[state_index] * GSF.weights[mdl_idx];
+		}
+		yaw_vector[0] += GSF.weights[mdl_idx] * cosf(EKF[mdl_idx].X[2]);
+		yaw_vector[1] += GSF.weights[mdl_idx] * sinf(EKF[mdl_idx].X[2]);
+	}
+	GSF.state[2] = atan2f(yaw_vector[1],yaw_vector[0]);
+
+	/*
+	// calculate a composite covariance matrix from a weighted average of the covariance for each model
+	// models with larger innovations are weighted less
+	memset(&P_GSF, 0, sizeof(P_GSF));
+	for (uint8_t mdl_idx = 0; mdl_idx < N_MODELS_EKFGSF; mdl_idx ++) {
+		float Xdelta[3];
+		for (uint8_t row = 0; row < 3; row++) {
+			Xdelta[row] = EKFGSF_mdl[mdl_idx].X[row] - X_GSF[row];
+		}
+		for (uint8_t row = 0; row < 3; row++) {
+			for (uint8_t col = 0; col < 3; col++) {
+				P_GSF[row][col] +=  EKFGSF_GSF.weights[mdl_idx] * (EKFGSF_mdl[mdl_idx].P[row][col] + Xdelta[row] * Xdelta[col]);
+			}
+		}
+	}
+	*/
+    GSF.yaw_variance = 0.0f;
```

Comment:
mixture of spaces and tabs. Run Tools/CodeStyle/ardupilot-astyle.sh libraries/AP_NavEKF/EKFGSF_yaw.{cpp,h}